File: Data/src/com/dlgdev/pokemon/database/Pokemon.java
package com.dlgdev.pokemon.database;

import java.util.HashMap;
import java.util.Map;

public class Pokemon {
	final int dexNumber;
	final int formNumber;
	public String name;
	public Map<String, String> names = new HashMap<>();
	public int[] baseStats = new int[6];
	public String[] types = new String[3];
	public double[] genderRatio = new double[2];
	public Ability[] abilities = new Ability[3];
	public String[] eggGroups = new String[2];
	public double height;
	public double weight;
	public String color;
	public String[] evos;
	public String prevo;
	public String[] otherForms;
	public String gender;
	public int evoLevel;
	public String baseForm;
	public String formName;
	public String formLetter;


	public Pokemon(int dexNumber, int formNumber) {
		this.dexNumber = dexNumber;
		this.formNumber = formNumber;
	}
}

File: database/src/com/dlgdev/pokemon/database/Pokemon.kt
package com.dlgdev.pokemon.database

import java.util.*

class Pokemon(val dexNumber: Int, val formNumber: Int) {
    var name: String = ""
    var names: MutableMap<String, String> = HashMap()
    var baseStats = IntArray(6)
    var types = arrayOfNulls<String>(3)
    var genderRatio = DoubleArray(2)
    var abilities = arrayOfNulls<com.dlgdev.pokemon.database.Ability>(3)
    var eggGroups = arrayOfNulls<String>(2)
    var height: Double = 0.toDouble()
    var weight: Double = 0.toDouble()
    var color: String? = null
    var evos = emptyArray<String?>()
    var prevo: String? = null
    var otherForms = emptyArray<String?>()
    var gender: String? = null
    var evoLevel: Int = 0
    var baseForm: String? = null
    var formName: String? = null
    var formLetter: String? = null
}
